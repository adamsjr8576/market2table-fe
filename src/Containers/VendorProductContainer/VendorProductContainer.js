import React, { useEffect } from 'react';
import { useSelector } from 'react-redux';
import PropTypes from 'prop-types';
import './VendorProductContainer.scss';
import VendorProductForm from '../VendorProductForm/VendorProductForm';
import VendorProductCard from '../VendorProductCard/VendorProductCard';

export const VendorProductContainer = ({ products, setProducts }) => {
  const { selectedVendor } = useSelector(state => state);

  if (selectedVendor.products && !products.length) {
    setProducts(selectedVendor.products);
  }

  let productCards;
  if (products.length) {
    productCards = products.map(product => {
      return (<VendorProductCard products={products} setProducts={setProducts} name={product.name} price={product.price} description={product.description} id={product.id} key={product.id}/>)
    });
  } else {
    productCards = (<p>You do not have any products on your account!</p>)
  }

  return (
    <section className='vendor-product-form-section'>
      <h2 className='vendor-product-form-title'>Add Vendor Product</h2>
      <VendorProductForm products={products} setProducts={setProducts}/>
      <section className='product-cards-container'>
        {productCards}
      </section>
    </section>
  )
}

export default VendorProductContainer;
