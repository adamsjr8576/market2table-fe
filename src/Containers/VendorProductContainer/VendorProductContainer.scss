.vendor-product-form-section {
  width: 40%;
  min-width: 325px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;
  height: 85%;
}

.vendor-product-form-title {
  width: 100%;
  text-align: center;
}

.product-cards-container {
  height: 50%;
  overflow: auto;
  margin-top: 10px;
  border: 2px solid black;
  border-radius: 10px;
  width: 95%;
}
