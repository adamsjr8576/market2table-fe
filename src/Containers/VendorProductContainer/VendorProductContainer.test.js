import React from 'react';
import { shallow } from 'enzyme';
import { VendorProductContainer } from './VendorProductContainer';

const mockState = {
  selectedVendor: {
    name: 'Rolling in the dough',
    description: 'just a little bakery rolling in some dough',
    image_link: 'rollthatdough.png',
    products: [{name: 'plain ol dough', price: '5', description: 'some good white bread', id: '2342'}]
  }
}

jest.mock("react-redux", () => ({
  useSelector: () => mockState
}));

describe('VendorProductContainer', () => {

  let wrapper;
  let mockProducts;
  let mocksetProducts = jest.fn();

  it('should match the snapshot with products', () => {
    mockProducts = [{name: 'plain ol dough', price: '5', description: 'some good white bread', id: '2342'}];
    wrapper = shallow(<VendorProductContainer
      products={mockProducts}
      setProducts={mocksetProducts}
    />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot without products', () => {
    mockProducts = [];
    wrapper = shallow(<VendorProductContainer
      products={mockProducts}
      setProducts={mocksetProducts}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
