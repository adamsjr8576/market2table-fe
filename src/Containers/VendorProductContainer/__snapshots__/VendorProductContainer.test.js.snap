// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VendorProductContainer should match the snapshot with products 1`] = `
<section
  className="vendor-product-form-section"
>
  <h2
    className="vendor-product-form-title"
  >
    Add Vendor Product
  </h2>
  <VendorProductForm
    products={
      Array [
        Object {
          "description": "some good white bread",
          "id": "2342",
          "name": "plain ol dough",
          "price": "5",
        },
      ]
    }
    setProducts={[MockFunction]}
  />
  <section
    className="product-cards-container"
  >
    <VendorProductCard
      description="some good white bread"
      id="2342"
      key="2342"
      name="plain ol dough"
      price="5"
      products={
        Array [
          Object {
            "description": "some good white bread",
            "id": "2342",
            "name": "plain ol dough",
            "price": "5",
          },
        ]
      }
      setProducts={[MockFunction]}
    />
  </section>
</section>
`;

exports[`VendorProductContainer should match the snapshot without products 1`] = `
<section
  className="vendor-product-form-section"
>
  <h2
    className="vendor-product-form-title"
  >
    Add Vendor Product
  </h2>
  <VendorProductForm
    products={Array []}
    setProducts={[MockFunction]}
  />
  <section
    className="product-cards-container"
  >
    <p>
      You do not have any products on your account!
    </p>
  </section>
</section>
`;
